client/redux/features/user/selectors.ts

```diff
@@ -0,0 +1,6 @@
+import { apiSlice } from "@/redux/features/api/slice";
+
+// 🎯 Selector function, allowing us to retrieve a value from the Redux state.
+
+// Calling `someEndpoint.select(someArg)` generates a new selector that will return the query result object for a query with those parameters
+export const selectUserData = apiSlice.endpoints.getUsers.select(null);
```

Comment:
How is this different from the getUsers hook?

Reply:
The createApi endpoint definition getUsers, exposed as useGetUsersQuery, is responsible for defining the query and managing the data fetching process, while the selector selectUserData is responsible for selecting and providing access to the fetched data within components. 
I have been remiss not to have included an example of using the selector after the hook dispatch.